core-bundle/src/Resources/contao/library/Contao/Controller.php

```diff
@@ -1762,7 +1764,7 @@ public static function addImageToTemplate($objTemplate, $arrItem, $intMaxWidth=n
 						}
 						catch (\Exception $e)
 						{
-							$objTemplate->$strHrefKey = static::addFilesUrlTo(System::urlEncode($arrItem['imageUrl']));
+							$objTemplate->$strHrefKey = static::addFilesUrlTo($arrItem['imageUrl']);
```

Comment:
I have removed the call to `System::urlEncode` here as it seems wrong to me. `$arrItem['imageUrl']` _must_ already be URL encoded, if a user enters a relative or absolute URL in the back end. If the content is an insert tag, then it _must_ be assumed that the insert tag outputs url encoded content (like the `{{file::*}}` insert tag does).

Reply:
Not sure about that. We generally should not remove code just because "it seems wrong". At least search the history and check why the code has been added.

Reply:
Reproduction:

* Put a file called `foo bar.jpg` into the `web/` folder.
* Create a new image content element.
* Enable **Overwrite meta data**.
* Set the **Image link target** to the publicly available file under `web/`, i.e. `foo%20bar.jpg`.
* Open the front end.

Generated output:

```html
<figure class="image_container">
  <a href="foo%2520bar.jpg" data-lightbox="eafe6b">
    <img src="…" alt="" itemprop="image">
  </a>
</figure>
```

The **Image link target** is erroneously double encoded due to

https://github.com/contao/contao/blob/b8f28f3fd33c52e6f0095deb7bebf431bd2a0bb0/core-bundle/src/Resources/contao/library/Contao/Controller.php#L1765